framework/src/controller/ipc/ipc_server.ts

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright © 2020 Lisk Foundation
+ *
+ * See the LICENSE file at the top-level directory of this distribution
+ * for licensing information.
+ *
+ * Unless otherwise agreed in a custom licensing agreement with the Lisk Foundation,
+ * no part of this software, including this file, may be copied, modified,
+ * propagated, or distributed except according to the terms contained in the
+ * LICENSE file.
+ *
+ * Removal or modification of this copyright notice is prohibited.
+ */
+
+// eslint-disable-next-line
+/// <reference path="../../external_types/pm2-axon/index.d.ts" />
+// eslint-disable-next-line
+/// <reference path="../../external_types/pm2-axon-rpc/index.d.ts" />
+
+import * as axon from 'pm2-axon';
+import { PubSocket, PullSocket } from 'pm2-axon';
+import { IPCSocket } from './ipc_socket';
+
+export class IPCServer extends IPCSocket {
+	public constructor(options: { socketsDir: string; name: string }) {
+		super(options);
+
+		this.pubSocket = axon.socket('pub', {}) as PubSocket;
+		this.subSocket = axon.socket('pull', {}) as PullSocket;
+	}
+
+	public async start(): Promise<void> {
+		await super.start();
```

Comment:
I'm not sure but if this server is running, should we handle errors or listen to error event on the server and handle or log it?

Reply:
I agree that we should add more logging into it at the debug level. I will create another issue an refer it here.

Reply:
@ishantiw I had set `CONNECTION_TIME_OUT=2000` which is the total time for connection including all tries. Each try time out default is 100ms and max try timeout is `5000`, but with our enforcement, it will reject in 2000. 

This connection timeout is for the clients when trying to connect to the server. But for server, its feasible as server don't need to connect to any. If it faces any issues opening the port or the socket, it will fail immediately. Suggest if we need to improve it further.

Reply:
okay, I think as of now `CONNECTION_TIME_OUT=2000` is good enough including total tries, I thought the socket only has a chance within 100ms to connect which can fail sometimes on Jenkins that's why I thought it should be increased. If there will be some failing tests later then we can rethink to change `2000` but at the moment it's good 👍